src/libraries/System.Memory.Data/tests/BinaryDataTests.cs

```diff
@@ -21,39 +21,236 @@ public partial class BinaryDataTests
         public void CanCreateBinaryDataFromBytes()
         {
             byte[] payload = "some data"u8.ToArray();
-            BinaryData data = BinaryData.FromBytes(payload);
-            Assert.Equal(payload, data.ToArray());
 
-            MemoryMarshal.TryGetArray<byte>(payload, out ArraySegment<byte> array);
-            Assert.Same(payload, array.Array);
+            {
+                BinaryData data = new BinaryData(payload);
```

Comment:
Would a [Theory] allow you to avoid all this repetition (and cause them to be separately reported too)

Reply:
Addressed in https://github.com/dotnet/runtime/pull/89605